crates/commitlog/src/segment.rs

```diff
@@ -573,6 +601,78 @@ impl Metadata {
 
         Ok(sofar)
     }
+
+    /// Finds the last valid commit in the segment using the offset index.
+    /// It traverses the index in reverse order, starting from the last key.
```

Comment:
This assume there could be more than last commit rather a tail of commits could be erased or corrupted. Did I went overbaord here?

Reply:
You're absolutely right.